compiler/src/dotty/tools/dotc/parsing/Scanners.scala

```diff
@@ -1399,8 +1399,8 @@ object Scanners {
 
   object IndentWidth {
     private inline val MaxCached = 40
-    private val spaces = Array.tabulate(MaxCached + 1)(new Run(' ', _))
-    private val tabs = Array.tabulate(MaxCached + 1)(new Run('\t', _))
+    private val spaces = Array.tabulate[Run](MaxCached + 1)(new Run(' ', _)) // TODO: remove new after bootstrap
```

Comment:
In fact, this gives me pause. Is this really what we want? I find it surprising and unnatural that one needs to add `[Run]` here. /cc @smarter

Reply:
An alternative design would be more local: Instead of  widening enum cases everywhere where they take part in inference we only decide at the application itself. I.e. `new Run(...)` is always a `Run` and will stay that way. `Run(...)` is of type `IndentWidth` unless its expected type is `Run`. We should evaluate which scheme is preferable. My tendency is to prefer the alternative.

Reply:
Sure. I mean the worrying thing is that the `Run` type is in a sense provisional, and can be revoked at each type inference step. That's analogous to union and singleton types of course. But I would argue that the behavior in these cases can also be suprising and it's a necessary tradeoff. Only for enum the tradeoff does not look to be so necessary or natural.